lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,63 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+        VALID_USER_HEADER_KEYS = %w(account_number user username email is_org_admin)
+
+        def self.with_logged_in_user(user = nil)
+          yield Thread.current[:attr_current_user] = new(user)
+        ensure
+          Thread.current[:attr_current_user] = nil
+        end
+
+        def initialize(user = nil)
+          @current = user || ManageIQ::API::Common::Headers.current
```

Comment:
This reads funny... I read this as "user or headers", which aren't related at all.  Is this being initialized with header data or a user object?